supervisor/install/install_linux.go

```diff
@@ -18,15 +18,108 @@
 package install
 
 import (
-	"errors"
+	"bytes"
+	"text/template"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/takama/daemon"
 )
 
-// Install installs service for linux. Not implemented yet.
+const daemonName = "myst_supervisor"
+const description = "Myst Supervisor"
+const descriptor = `
+[Unit]
+Description={{.Description}}
+Documentation=https://mysterium.network/
+
+[Service]
+PIDFile=/run/{{.Name}}.pid
+ExecStartPre=/bin/rm -f /run/{{.Name}}.pid
+ExecStart={{.Path}} {{.Args}}
+Restart=on-failure
+
+[Install]
+WantedBy=multi-user.target
+`
+
+// Install installs service for linux.
 func Install(options Options) error {
-	return errors.New("not implemented")
+	if !options.valid() {
+		return errInvalid
+	}
+
+	log.Info().Msg("Configuring Myst Supervisor")
+	dmn, err := mystSupervisorDaemon(options)
+	if err != nil {
+		return err
+	}
+
+	log.Info().Msg("Cleaning up previous installation")
+	clean(dmn)
+
+	log.Info().Msg("Installing daemon")
+	err = execAndLog(func() (string, error) { return dmn.Install() })
+	if err != nil {
+		return err
+	}
+
+	err = execAndLog(dmn.Start)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func mystSupervisorDaemon(options Options) (daemon.Daemon, error) {
+	dmn, err := daemon.New(daemonName, description, daemon.SystemDaemon)
+	if err != nil {
+		return nil, err
+	}
+
+	t, err := template.New("unit-descriptor").Parse(description)
+	if err != nil {
+		return nil, err
+	}
+	buffer := new(bytes.Buffer)
+	err = t.Execute(buffer, map[string]string{
```

Comment:
Where this template used? It's just created and never reused or I'm missing something?

Reply:
yes, a bug - lost some changes with stashing